src/cljs/webchange/editor_v2/concepts/views.cljs

```diff
@@ -31,60 +29,55 @@
-              dataset-id (:dataset-id @data)]
-          [ui/card
-           [ui/card-header {:title "Add dataset item"}]
-           [ui/card-content
-            [ui/select {:value dataset-id :on-change #(swap! data assoc :dataset-id (-> % .-target .-value))}
-             (for [dataset datasets]
-               [ui/menu-item {:value (:id dataset)} (:name dataset)])]
-            [ui/text-field {:label "Name" :full-width true :default-value (:name @data) :on-change #(swap! data assoc :name (-> % .-target .-value))}]
-            [ui/divider]
-            [dataset-item-fields-panel dataset-id data]]
-           [ui/card-actions
-            [ui/button {:on-click #(re-frame/dispatch [::concepts-events/add-dataset-item @data])} "Add"]
-            [ui/button {:on-click #(redirect-to :course-editor-v2 :id course-id)} "Cancel"]
-            (when (:add-dataset-item loading)
-              [ui/circular-progress])]
-           ]))
-      [ui/circular-progress]
-      )))
+      [ui/card
+       [ui/card-content
+        [ui/select {:value dataset-id
+                    :on-change #(swap! data assoc :dataset-id (-> % .-target .-value))
+                    :style {:min-width "150px"}}
+         (for [dataset datasets]
+           [ui/menu-item {:value (:id dataset)} (:name dataset)])]
+        [ui/text-field {:label "Name" :full-width true :default-value (:name @data) :on-change #(swap! data assoc :name (-> % .-target .-value))}]
+        [ui/divider]
+        [dataset-item-fields-panel dataset-id data]]
+       [ui/card-actions
+        [ui/button {:on-click #(re-frame/dispatch [::concepts-events/add-dataset-item @data])} "Add"]
+        [ui/button {:on-click #(redirect-to :course-editor-v2 :id course-id)} "Cancel"]
+        (when (:add-dataset-item loading)
+          [ui/circular-progress])]]
+      [ui/circular-progress])))
 
 (defn edit-dataset-item-form
   [course-id item-id]
   (let [item @(re-frame/subscribe [::concepts-subs/dataset-item item-id])
+        loading @(re-frame/subscribe [:loading])
         dataset-id (:dataset-id item)
         data (r/atom {:data (:data item)
                       :name (:name item)})]
     (if item
-      (fn []
-        (let [loading @(re-frame/subscribe [:loading])]
-          [ui/card
-           [ui/card-header {:title "Edit dataset item"}]
-           [ui/card-content
-            [ui/text-field {:label "Name" :full-width true :default-value (:name @data) :on-change #(swap! data assoc :name (-> % .-target .-value))}]
-            [ui/divider]
-            [dataset-item-fields-panel dataset-id data]]
-           [ui/card-actions
-            [ui/button {:on-click #(re-frame/dispatch [::concepts-events/edit-dataset-item item-id @data])} "Save"]
-            [ui/button {:on-click #(redirect-to :course-editor-v2 :id course-id)} "Cancel"]
-            (when (:edit-dataset-item loading)
-              [ui/circular-progress])]
-           ]))
-      [ui/circular-progress]
-      )))
+      [ui/card
```

Comment:
Make sure this form works when you reload the page. An 'item' might not be loaded correctly and 'data' wouldn't be initialized properly.